src/umpire/ResourceManager.cpp

```diff
@@ -565,21 +565,8 @@ ResourceManager::reallocate(void* src_ptr, std::size_t size, Allocator allocator
 #if defined(UMPIRE_ENABLE_NUMA)
 static strategy::NumaPolicy* cast_as_numa_policy(Allocator& allocator) {
```

Comment:
Why don't we just use `unwrap_allocator` directly (I'm unsure what this function is helping with?)

Reply:
`unwrap_allocator` throws an error if it can't cast as that type